waku/v2/protocol/relay/waku_relay.go

```diff
@@ -227,9 +227,38 @@ func (w *WakuRelay) Subscribe(ctx context.Context, topic *Topic) (*Subscription,
 	return subscription, nil
 }
 
+func (w *WakuRelay) Unsubscribe(ctx context.Context, topic *Topic) error {
+	if _, ok := w.topics[*topic]; !ok {
+		return fmt.Errorf("topics %s is not subscribed", (string)(*topic))
+	}
+	log.Info("Unsubscribing from topic ", *topic)
+	delete(w.topics, *topic)
+
+	for _, sub := range w.subscriptions[*topic] {
+		sub.Unsubscribe()
+	}
+
+	w.relaySubs[*topic].Cancel()
+	delete(w.relaySubs, *topic)
+
+	err := w.wakuRelayTopics[*topic].Close()
+	if err != nil {
+		return err
+	}
+	delete(w.wakuRelayTopics, *topic)
+
+	return nil
+}
```

Comment:
What do you think if instead of receiving a pointer to a `Topic`,  we receive the `Topic` directly?
I also added the topics and subscription mutex:

```suggestion
func (w *WakuRelay) Unsubscribe(ctx context.Context, topic Topic) error {
	w.topicsMutex.Lock()
	defer w.topicsMutex.Unlock()

	if _, ok := w.topics[topic]; !ok {
		return fmt.Errorf("topics %s is not subscribed", topic)
	}
	log.Info("Unsubscribing from topic ", topic)
	delete(w.topics, topic)

	w.subscriptionsMutex.Lock()
	for _, sub := range w.subscriptions[topic] {
		sub.Unsubscribe()
	}
	subscriptionsMutex.Unlock()

	w.relaySubs[topic].Cancel()
	delete(w.relaySubs, topic)

	err := w.wakuRelayTopics[topic].Close()
	if err != nil {
		return err
	}
	delete(w.wakuRelayTopics, topic)

	return nil
}
```